File: chess_cpp/validators/EndSlotWithinBorderValidator.cpp
//
// Created by Hrach Khachatryan on 12/15/19.
//

#include "EndSlotWithinBorderValidator.h"
#include "../utils/MessageHandler.h"

bool EndSlotWithinBorderValidator::validate(Board *board, Spot *start, Spot *end) {
    return end->getX() >= 0 && end->getX() <= 7 && end->getY() >= 0 && end->getY() <= 7;
}

std::string EndSlotWithinBorderValidator::getNotValidMessage() {
    return MessageHandler::getHellNoMessage();
}

File: chess_java/src/chess/validators/EndSlotWithinBorderValidator.java
package chess.validators;

import chess.Board;
import chess.Spot;
import chess.utils.MessageHandler;

public class EndSlotWithinBorderValidator extends PieceMoveValidator {
    @Override
    public boolean validate(Board board, Spot start, Spot end) {
        return end.getX() >= 0 && end.getX() <= 7 && end.getY() >= 0 && end.getY() <= 7;
    }

    @Override
    protected String getNotValidMessage() {
        return MessageHandler.getHellNoMessage();
    }
}
